zebra-chain/src/sapling/tree.rs

```diff
@@ -108,71 +108,134 @@ impl From<&Root> for [u8; 32] {
     }
 }
 
-/// Sapling Note Commitment Tree
+/// Sapling Incremental Note Commitment Tree
 #[derive(Clone, Debug, Default, Eq, PartialEq)]
 struct NoteCommitmentTree {
-    /// The root node of the tree (often used as an anchor).
-    root: Root,
-    /// The height of the tree (maximum height for Sapling is 32).
-    height: u8,
-    /// The number of leaves (note commitments) in this tree.
-    count: u32,
+    left: Option<[u8; 32]>,
+    right: Option<[u8; 32]>,
+    parents: Vec<Option<[u8; 32]>>,
```

Comment:
Types: Should these be a `sapling::Hash` instead?

Reply:
Unlike Orchard, the Sapling Merkle hash function has byte buffers as inputs and outputs (even though they seem to be the enconding of point coordinates too). I'm not sure the exact reason but it's better to follow the spec

Reply:
Let's add a short comment explaining the difference from our standard Zebra types.

Reply:
Added in c3e556d2